sdk/keyvault/azure-security-keyvault-common/src/keyvault_pipeline.cpp

```diff
@@ -70,8 +70,7 @@ std::unique_ptr<Azure::Core::Http::RawResponse> _internal::KeyVaultPipeline::Sen
     case Azure::Core::Http::HttpStatusCode::NoContent:
       break;
     default:
-      throw Azure::Security::KeyVault::_internal::KeyVaultException::CreateException(
-          std::move(response));
+      throw Azure::Core::RequestFailedException(response);
```

Comment:
I don't think this is a good idea. This means that Azure Core needs to know how to parse a Key Vault Keys service error response, or ignore any specific Key Vault Keys error information beyond the RequestFailException data

Reply:
Is there anything in the implementation of `CreateException` that is KeyVault specific?

Reading the `error` and `message` properties from JSON seemed general purpose.

Reply:
I am not sure if all Azure services uses the same criteria in the error response for:
- Json structure;  the `error` and `message` field could be in some deeper level in the json and not in the root.
- field names; a service might use `err`, or any other name for the fields.